File: client/python/mesh/test/macro/proxy_test.py
#
# Copyright (c) 2000, 2023, trustbe and/or its affiliates. All rights reserved.
# TRUSTBE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
#
#
import mesh.log as log
import types
import unittest
from abc import ABC, abstractmethod
from typing import Any

from mesh.macro import Proxy, InvocationHandler


class Foo(ABC):

    @abstractmethod
    def bar(self) -> str:
        pass


class ProxyHandler(InvocationHandler):

    def invoke(self, proxy: Any, func: types.FunctionType, *args, **kwargs):
        return "bar"


class TestProxy(unittest.TestCase):

    def test_get_environ(self):
        foo = Proxy.new_proxy(Foo, ProxyHandler())
        bar = foo.bar()
        log.info(bar)


if __name__ == '__main__':
    unittest.main()

File: client/java/src/test/java/com/be/mesh/client/mpc/ProxyTest.java
/*
 * Copyright (c) 2000, 2023, trustbe and/or its affiliates. All rights reserved.
 * TRUSTBE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
 *
 *
 */
package com.be.mesh.client.mpc;

import com.be.mesh.client.prsim.Cache;
import lombok.extern.slf4j.Slf4j;
import org.testng.annotations.Test;

import java.util.Map;

/**
 * @author coyzeng@gmail.com
 */
@Slf4j
public class ProxyTest {

    @Test
    public void testProxy() {
        Cache cache = ServiceProxy.proxy(Cache.class);
        Map<String, Object> ret = cache.get("x", Types.MapObject);
        log.info("{}", ret);
    }

    @Test
    public void testProxyDefault() {
        Cache cache = ServiceProxy.proxy(Cache.class);
        String v = cache.get("t", Types.of(String.class));
        log.info(v);
    }
}
